tests/test_integration/test_tag_filter.py

```diff
@@ -0,0 +1,142 @@
+import os
+import subprocess
+import textwrap
+import time
+from datetime import datetime, timedelta
+
+import pytest
+
+from tests.lib.util import (
+    create_commit,
+    create_file,
+    create_tag,
+    get_sha,
+    get_version,
+    get_version_module,
+    get_version_script,
+)
+
+pytestmark = pytest.mark.all
+
+
+@pytest.mark.important
+@pytest.mark.parametrize(
+    "tag_filter, version",
+    [
+        ("product_x/(?P<tag>.*)", "1.1.0"),
+        ("product_y/(?P<tag>.*)", "1.1.10"),
+        ("product_z/foo/(?P<tag>.*)", "1.1.1"),
+    ],
+)
+def test_tag_filter(repo, create_config, tag_filter, version):
+    if tag_filter:
+        create_config(repo, {"tag_filter": tag_filter, "tag_formatter": tag_filter})
+    else:
+        create_config(repo)
+
+    commits = {}
+    tags_to_commit = [
+        "product_x/1.0.0",
+        "product_x/1.0.2",
+        "product_x/1.1.0",
+        "product_y/1.1.10",
+        "product_z/foo/1.1.1",
+    ]
+
+    for i, tag in enumerate(tags_to_commit):
+        create_file(repo, commit=False)
+        dt = datetime.now() - timedelta(days=len(tags_to_commit) - i)
+        create_commit(repo, "Some commit", dt=dt)
+        commits[tag] = get_sha(repo)
+        time.sleep(1)
+
+    tags_to_create = [
+        "product_x/1.0.0",
+        "product_x/1.0.2",
+        "product_x/1.1.0",
+        "product_y/1.1.10",
+        "product_z/foo/1.1.1",
+    ]
+
+    for tag in tags_to_create:
+        create_tag(repo, tag, message="", commit=commits[tag])
```

Comment:
Why not create tag immediately after the commit?

Reply:
I was just following the formatter test structure.